packages/@aws-cdk/pipelines/README.md

```diff
@@ -0,0 +1,542 @@
+  synthAction: SimpleSynthAction.standardNpmSynth({
+    sourceArtifact,
+    cloudAssemblyArtifact,
+    buildCommand: 'npm run build',
+    additionalArtifacts: [
+      {
+        directory: 'test',
+        artifact: integTestsArtifact,
+      }
+    ],
+  }),
+  // ...
+});
+
+const validationAction = new ShellScriptAction({
+  name: 'TestUsingBuildArtifact',
+  additionalArtifacts: [integTestsArtifact],
+  // 'test.js' was produced from 'test/test.ts' during the synth step
+  commands: ['node ./test.js'],
+});
+```
+
+## CDK Bootstrapping
+
+An *environment* is an *(account, region)* pair where you want to deploy a CDK
+stack (see [Environments](https://docs.aws.amazon.com/cdk/latest/guide/environments.html)
+in the CDK Developer Guide).
+
+Before you can provision the pipeline, you have to *bootstrap* the environment you want
+to create it in. If you are deploying your application to different environments, you
+also have to bootstrap those and be sure to add a *trust* relationship.
+
+To bootstrap an environment for provisioning the pipeline:
+
+```
+$ env CDK_NEW_BOOTSTRAP=1 npx cdk bootstrap \
+    [--profile admin-profile-1] \
+    --cloudformation-execution-policies arn:aws:iam::aws:policy/AdministratorAccess \
+    aws://111111111111/us-east-1
+```
+
+To bootstrap a different environment for deploying CDK applications into using
+a pipeline in account `111111111111`:
+
+```
+$ env CDK_NEW_BOOTSTRAP=1 npx cdk bootstrap \
+    [--profile admin-profile-2] \
+    --cloudformation-execution-policies arn:aws:iam::aws:policy/AdministratorAccess \
+    --trust 11111111111 \
+    aws://222222222222/us-east-2
+```
+
+These command lines explained:
+
+* `npx`: means to use the CDK CLI from the current NPM install. If you are using
+  a global install of the CDK CLI, leave this out.
+* `--profile`: should indicate a profile with administrator privileges that has
+  permissions to provision a pipeline in the indicated account. You can leave this
+  flag out if either the AWS default credentials or the `AWS_*` environment
+  variables confer these permissions.
+* `--cloudformation-execution-policies`: ARN of the managed policy that future CDK
+  deployments should execute with. You can tailor this to the needs of your organization
+  and give more constrained permissions than `AdministratorAccess`.
+* `--trust`: indicates which other account(s) should have permissions to deploy
+  CDK applications into this account. In this case we indicate the Pipeline's account,
+  but you could also use this for developer accounts (don't do that for production
+  application accounts though!).
+* `aws://222222222222/us-east-2`: the account and region we're bootstrapping.
+
+> **Security tip**: we recommend that you use administrative credentials to an
+> account only to bootstrap it and provision the initial pipeline. Otherwise,
+> access to administrative credentials should be dropped as soon as possible.
+
```

Comment:
Add a mention and a reference here to the bootstrap cfn template and indicate that it can be used through stack sets or control tower to bootstrap within the Organizations

You should also indicate that the bootstrap environment can be customized and at least offer a link to the code that shows the various degrees of freedom and say that the template should match the synthesized output

Reply:
Does the name of the CDK stack not matter? I thought it needs to be CDKToolkit or you have to provide the name on every deploy using the toolkit-stack-name arg.

Reply:
It does matter for the CLI but it does not matter for this library.

Reply:
So when you use a different StackName, the cross-account CI/CD story works but you cannot just use the CLI in the child account anymore without specifying the stack name? Is this correct?

Reply:
That's right.